src/main/kotlin/io/kotest/assertions/arrow/option/matchers.kt

```diff
@@ -3,61 +3,105 @@ package io.kotest.assertions.arrow.option
 import arrow.core.None
 import arrow.core.Option
 import arrow.core.Some
+import io.kotest.assertions.arrow.core.shouldBeNone
+import io.kotest.assertions.arrow.core.shouldBeSome
 import io.kotest.matchers.Matcher
 import io.kotest.matchers.MatcherResult
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldNot
 import kotlin.contracts.ExperimentalContracts
 import kotlin.contracts.contract
 
+@Deprecated(
+  "Use shouldBeSome from core",
+  ReplaceWith("shouldBeSome()", "io.kotest.assertions.arrow.core.shouldBeSome")
+)
 @OptIn(ExperimentalContracts::class)
 fun Option<*>.shouldBeSome() {
   contract {
     returns() implies (this@shouldBeSome is Some<*>)
   }
-  this should beSome()
+  shouldBeSome()
 }
 
-fun beSome() = object : Matcher<Option<*>> {
-  override fun test(value: Option<*>): MatcherResult =
-      MatcherResult(value is Some, "$value should be Some", "$value should not be Some")
+@Deprecated(
+  "Use shouldBNone from core",
+  ReplaceWith("shouldBeNone()", "io.kotest.assertions.arrow.core.shouldBeNone")
+)
+@OptIn(ExperimentalContracts::class)
+fun Option<*>.shouldBeNone() {
+  contract {
+    returns() implies (this@shouldBeNone is Some<*>)
+  }
+  shouldBeNone()
 }
 
-infix fun <T> Option<T>.shouldBeSome(t: T) = this should beSome(t)
-infix fun <T> Option<T>.shouldNotBeSome(t: T) = this shouldNot beSome(t)
-fun <T> beSome(t: T) = object : Matcher<Option<T>> {
-  override fun test(value: Option<T>): MatcherResult {
-    return when (value) {
-      is None -> {
-        MatcherResult(false, "Option should be Some($t) but was None", "")
-      }
-      is Some -> {
-        if (value.value == t)
-          MatcherResult(true, "Option should be Some($t)", "Option should not be Some($t)")
-        else
-          MatcherResult(false, "Option should be Some($t) but was Some(${value.value})", "")
+@Deprecated("Use shouldBeSome directly")
+fun beSome(): Matcher<Option<*>> =
```

Comment:
I wouldn't deprecate these because some people use the syntax:

a shouldBe foo

where foo is like beSome()